### PR TITLE
Initial Implementation of LocalLink

### PR DESCRIPTION
**Pull request by tomitrescak**
- [x] If this PR is a new feature, reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests
- [ ] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass
- [x] Update CHANGELOG.md with your change
- [x] Add your name and email to the AUTHORS file (optional)
- [ ] If this was a change that affects the external API, update the docs and post a link to the PR in the discussion

As discussed in #66.

I could not figure out how to mock graphql import, so one more test is necessary. Docs probably could be improved as well.

### CODE REVIEW

File: packages/apollo-link-local/package.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
+  "name": "apollo-link-http",
```
Comment (fbartho) [inline packages/apollo-link-local/package.json]: The meta info here probably should be updated!

**Comment by intellix** [comment]
Apologies for my lack of knowledge of how Link, but I'm just wondering why you would need to pass in a schema to return local data? 

I thought the purpose of Link was just for the resolving of the data. In our Angular mocks we were previously doing something like:

```ts
return new ApolloClient({
  networkInterface: {
    query: query => new Promise(resolve => resolve({
      data: {
        hello: 'world',
      },
    })),
  },
});
```

Something like the following seems sufficient for mocking responses to me and was kind of what I was hoping for :)

```ts
import { ApolloLink, Operation, FetchResult, Observable } from 'apollo-link';

export class LocalLink extends ApolloLink {

  constructor(private data: any) {
    super();
  }

  public request(operation: Operation): Observable<FetchResult> | null {
    return new Observable<FetchResult>(observer => observer.next(this.data));
  }
}

export default LocalLink;
```

**Comment by jbaxleyiii** [comment]
@tomitrescak this looks great! Could you update the package info and instead of `LocalLink` call it something like `SchemaLink`?

File: packages/apollo-link-local/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+title: Http Link
```
Comment (zephraph) [inline packages/apollo-link-local/README.md]: Title here should be updated

**Comment by zephraph** [comment]
@jbaxleyiii could you speak a little to what SchemaLink conveys? It seems confusing to me because it seems the other names are how they link or what they do. SchemaLink doesn't quite tell that story, but maybe I'm thinking about it wrong. 

Also, what else needs to be done here? There are a lot of name references that need to be updated (more if the name changes) and it looks like a unit test needs to be finished out. Anything else?

**Comment by fbartho** [comment]
Alternative names?
- ShimLink
- MockLink
- VirtualLink

I think what @jbaxleyiii was getting at with `SchemaLink` was that we're querying, and the Schema itself is generating the response?

**Comment by zephraph** [comment]
I'm fine with the name. You're linking directly to an executable schema, so it makes sense. This can be closed in favor of #284.

**Comment by zephraph** [comment]
@jbaxleyiii we can close this one

**Comment by imranolas** [comment]
Haha, talk about multiple discovery. https://github.com/imranolas/apollo-link-local

**Comment by stubailo** [comment]
Great, closing!

**Comment by tomitrescak** [comment]
Just cam back from a week long conference. Am glad you guys pushed this forward. Great work. @imranolas isn't this now a clone of apollo-link-schema? What is the difference?

**Comment by imranolas** [comment]
@tomitrescak There is no difference hence the deprecation #316

**Pull request closed without merging**